src/spatial.jl

```diff
@@ -759,7 +759,9 @@ Not currently compatible with automatic differentiation using Zygote.
 function scale_coords!(sys, scale_factor; ignore_molecules=false)
     if ignore_molecules || isnothing(sys.topology)
         sys.boundary = scale_boundary(sys.boundary, scale_factor)
-        sys.coords = sys.coords .* scale_factor
+        for i in eachindex(sys.coords)
```

Comment:
This is slow on the GPU, unfortunately. You might have to write a function that does `coord .* scale_factor` and broadcast that function (like `ustrip_vec`, a double broadcast).

Also, does the branch with molecules below work when `scale_factor` is a `SVector`?

Reply:
I added a new `scale_vec` function:

```julia
"""
    scale_vec(v, c)
Broadcasted form of `.* c`, allowing scaling all the vectors in v by c, where c is
either a constant or a vector.
"""
scale(v, c) = v .* c
scale_vec(v, c) = scale.(v, Ref(c))
```

and used it in `scale_coords!`.

I also refactored the branch used when a topology is present:

```julia
# Move all atoms in a molecule by the same amount according to the molecule center
# Then move the atoms back to the molecule center and wrap in the scaled boundary
shift_vecs = scale_vec(mol_centers, scale_factor .- 1)
```